src/pages/export/Event.js

```diff
@@ -1,197 +1,76 @@
-                    paging: false,
-                    fields: 'id,path,displayName,children::isNotEmpty',
-                })
-                .then(root => root.toArray()[0])
-
-            this.setState({
-                orgUnit: {
-                    selected: [],
-                    value: orgUnitTree,
-                },
-            })
-        } catch (e) {
-            !isProduction && console.log('fetch Programs failed')
-            !isProduction && console.log(e)
-        }
-    }
-
-    async fetchProgramStages(id) {
-        try {
-            const {
-                data: { programStages },
-            } = await api.get(
-                `programs/${id}.json?fields=id,displayName,programStages[id,displayName]`
-            )
-            const values = programStages.map(({ id, displayName }) => ({
-                value: id,
-                label: displayName,
-            }))
-
-            values.unshift({
-                value: -1,
-                label: i18n.t('[ All program stages]'),
-            })
-            const selected = values[0]['value']
-            this.setState({
-                programStages: { values, selected },
-            })
-        } catch (e) {
-            !isProduction && console.log('fetch ProgramStages failed', id)
-            !isProduction && console.log(e)
-        }
-    }
-
-    onFormUpdate = (name, value) => {
-        if (name === 'programs') {
-            this.fetchProgramStages(value)
-        }
-    }
-
-    onSubmit = () => {
-        const {
-            orgUnit,
-            startDate,
-            endDate,
-            programStages,
-            inclusion,
-            format,
-            compression,
-        } = this.getFormState()
-
-        let attachment = `events${format}`
-        if (compression !== 'none') {
-            attachment += compression
-        }
-
-        const append = []
-
-        if (programStages !== -1) {
-            append.push(`programStage=${programStages}`)
-        }
-
-        if (orgUnit.length > 0) {
-            const path = orgUnit[0]
-            const orgUnitId = path.substr(path.lastIndexOf('/') + 1)
-            append.push(`orgUnit=${orgUnitId}`)
-        }
-
-        append.push('links=false')
-        append.push('skipPaging=true')
-        append.push(`attachment=${attachment}`)
-        append.push('startDate=' + moment(startDate).format('YYYY-MM-DD'))
-        append.push('endDate=' + moment(endDate).format('YYYY-MM-DD'))
-        append.push(`ouMode=${inclusion.toUpperCase()}`)
-        append.push(`format=${format.substr(1)}`)
-
-        let path = `events${format}`
-        if (compression !== 'none') {
-            path += `${compression}`
-        }
-
-        const params = getParamsFromFormState(
-            this.getFormState(),
-            ['programs', 'includeDeleted', 'idScheme'],
-            append
-        )
-
-        download(api.url(path) + '?' + params)
-    }
 }
+
+EventExport.dataTest = 'export-event'
```

Comment:
Stylewise it might be nice to not assign to the main export, but export them separately, like in ui-core. But if you agree, I'd be fine with dealing with that in a separate PR.

Reply:
Yeah, I'd prefer to do that in a dfferent PR